src/Init/Prelude.lean

```diff
@@ -1877,14 +1877,18 @@ def reservedMacroScope := 0
 def firstFrontendMacroScope := hAdd reservedMacroScope 1
 
 class MonadRef (m : Type → Type) where
-  getRef      : m Syntax
-  withRef {α} : Syntax → m α → m α
+  getRef             : m Syntax
+  withRef {α}        : Syntax → m α → m α
+  getElaborator      : m Name
+  withElaborator {α} : Name → m α → m α
```

Comment:
I didn't find any polymorphic function that uses `withElaborator`. It seems it is always used in concrete monads where `withReader` could be used.

Reply:
Yes, you are right. I think I mostly added it for consistency.
But even the polymorphic uses of `getElaborator` are rare.
* There is `resolveGlobalConst*`. Not sure if we really need it there; setting the elaborator to `Name.anonymous` would mean that e.g. on the backticks of a double quoted name you would see the elaborator's docstring, but not on the name itself. Since go-to-definition should definitely *not* go to the elaborator in the latter case, that might simply be more consistent.
* Similarly, it is used (monomorphically) by `addTermInfo`, but in every case I see the surrounding elaborator (almost always `app`) is not an interesting information in that place.
* Finally, there is `mkTacticInfo`; here it does save us from having to type out (and possibly mistype) the current tactic's name when we use it or `mkInitialInfo`.

So, I would suggest to revert the `MonadRef` changes and keep `elaborator` in the context for the tactic monad only. Sounds better?

Reply:
> So, I would suggest to revert the MonadRef changes and keep elaborator in the context for the tactic monad only. Sounds better?

Yes. Thanks.

Reply:
done